artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPConnectionContext.java

```diff
@@ -340,7 +344,19 @@ protected void remoteLinkOpened(Link link) throws Exception {
    }
 
    private boolean isReplicaTarget(Link link) {
-      return link != null && link.getTarget() != null && link.getTarget().getAddress() != null && link.getTarget().getAddress().equals(ProtonProtocolManager.MIRROR_ADDRESS);
+      boolean hasMirror = false;
+
+      Terminus terminus = (Terminus)link.getTarget();
+      if (terminus != null && terminus.getCapabilities() != null) {
+         for (Symbol s : terminus.getCapabilities()) {
+            if (s.equals(AMQPMirrorControllerSource.MIRROR_CAPABILITY)) {
+               hasMirror = true;
+               break;
+            }
+         }
+      }
+
+      return link != null && link.getTarget() != null && link.getTarget().getAddress() != null && hasMirror;
```

Comment:
Superfluous null check, as it would have NPE'd at the top of the method otherwise. The terminus also has a variable that could be used.

Reply:
you're right.. thanks